src/extension/flutter/vm_service_extensions.ts

```diff
@@ -52,13 +52,13 @@ export class VmServiceExtensions {
 	}
 
 	/// Handles an event from the Debugger, such as extension services being loaded and values updated.
-	public async handleDebugEvent(session: DartDebugSessionInformation, e: vs.DebugSessionCustomEvent): Promise<void> {
+	public async handleDebugEvent(session: DartDebugSessionInformation, e: vs.DebugSessionCustomEvent, forceFlutterWorkspace?: boolean): Promise<void> {
 		if (e.event === "dart.serviceExtensionAdded") {
 			this.handleServiceExtensionLoaded(session, e.body.extensionRPC as VmServiceExtension, e.body.isolateId as string | null | undefined);
 
 			try {
 				if (e.body.extensionRPC === VmServiceExtension.InspectorSetPubRootDirectories) {
-					const projectFolders = await getAllProjectFolders(this.logger, getExcludedFolders, { requirePubspec: true, searchDepth: config.projectSearchDepth });
+					const projectFolders = await getAllProjectFolders(this.logger, getExcludedFolders, { requirePubspec: !forceFlutterWorkspace, searchDepth: forceFlutterWorkspace ? 1 : config.projectSearchDepth, forceFlutterWorkspace });
```

Comment:
Are these arguments (ignoring pubspec etc.) specific to this call, or do we want to use them everywhere we look for projects? There are quite a few places that call `getAllProjectFolders` and I suspect they should all behave the same. If so (but we don't think we need to fix this now), we could add a TODO. Probably the fix is to make this function simpler so everyone that calls it doesn't need to provide all these arguments explicitly.